process/sync/shardblock.go

```diff
@@ -142,6 +142,78 @@ func NewShardBootstrap(
 	return &boot, nil
 }
 
+func (boot *ShardBootstrap) addHeaderToForkDetector(shardId uint32, nonce uint64, lastNotarizedMetaNonce uint64) {
+	header, headerHash, errNotCritical := boot.storageBootstrapper.getHeader(shardId, nonce)
+	if errNotCritical != nil {
+		log.Info(errNotCritical.Error())
+		return
+	}
+
+	shardHeader, ok := header.(*block.Header)
+	if !ok {
+		return
+	}
+
+	// search up from last notarized till shardHeader.Round
+	highestOwnHdrFromMetaChain := &block.Header{}
+	highestMetaNonceInStorer := lastNotarizedMetaNonce
+	for {
+		hdr, _, errNotCritical := boot.getMetaHeaderFromStorage(sharding.MetachainShardId, highestMetaNonceInStorer)
+		if errNotCritical != nil {
+			log.Info(errNotCritical.Error())
+			highestMetaNonceInStorer--
+			break
```

Comment:
before these breaks I think we might test if the hdr and meta hdr is really retrievable. Otherwise should continue to decrement.

Reply:
it is tested. Exactly that. You are trying to get the highest metablock which includes a block from your current shard.